include/openssl/kdf.h

```diff
@@ -15,80 +15,107 @@
+# define EVP_PKEY_CTRL_TLS_MD                   EVP_PKEY_CTRL_KDF_MD
+# define EVP_PKEY_CTRL_HKDF_MD                  EVP_PKEY_CTRL_KDF_MD
+# define EVP_PKEY_CTRL_HKDF_SALT                EVP_PKEY_CTRL_KDF_SALT
+# define EVP_PKEY_CTRL_HKDF_KEY                 EVP_PKEY_CTRL_KDF_KEY
+# define EVP_PKEY_CTRL_HKDF_INFO                EVP_PKEY_CTRL_KDF_INFO
+# define EVP_PKEY_CTRL_HKDF_MODE                EVP_PKEY_CTRL_KDF_MODE
 
 # define EVP_PKEY_HKDEF_MODE_EXTRACT_AND_EXPAND 0
 # define EVP_PKEY_HKDEF_MODE_EXTRACT_ONLY       1
 # define EVP_PKEY_HKDEF_MODE_EXPAND_ONLY        2
 
-# define EVP_PKEY_CTX_set_tls1_prf_md(pctx, md) \
-            EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
-                              EVP_PKEY_CTRL_TLS_MD, 0, (void *)(md))
-
-# define EVP_PKEY_CTX_set1_tls1_prf_secret(pctx, sec, seclen) \
-            EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
-                              EVP_PKEY_CTRL_TLS_SECRET, seclen, (void *)(sec))
 
-# define EVP_PKEY_CTX_add1_tls1_prf_seed(pctx, seed, seedlen) \
+/* Generic setters */
+# define EVP_PKEY_CTX_set_pbe_md(pctx, md) \
             EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
-                              EVP_PKEY_CTRL_TLS_SEED, seedlen, (void *)(seed))
+                              EVP_PKEY_CTRL_KDF_MD, 0, (void *)(md))
 
-# define EVP_PKEY_CTX_set_hkdf_md(pctx, md) \
+# define EVP_PKEY_CTX_set1_pbe_pass(pctx, pass, passlen) \
             EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
-                              EVP_PKEY_CTRL_HKDF_MD, 0, (void *)(md))
+                              EVP_PKEY_CTRL_KDF_PASS, passlen, (void *)(pass))
 
-# define EVP_PKEY_CTX_set1_hkdf_salt(pctx, salt, saltlen) \
+# define EVP_PKEY_CTX_set1_pbe_salt(pctx, salt, saltlen) \
             EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
-                              EVP_PKEY_CTRL_HKDF_SALT, saltlen, (void *)(salt))
+                              EVP_PKEY_CTRL_KDF_SALT, saltlen, (void *)(salt))
 
-# define EVP_PKEY_CTX_set1_hkdf_key(pctx, key, keylen) \
+# define EVP_PKEY_CTX_set1_pbe_key(pctx, key, keylen) \
             EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
-                              EVP_PKEY_CTRL_HKDF_KEY, keylen, (void *)(key))
+                              EVP_PKEY_CTRL_KDF_KEY, keylen, (void *)(key))
 
-# define EVP_PKEY_CTX_add1_hkdf_info(pctx, info, infolen) \
+# define EVP_PKEY_CTX_add1_pbe_info(pctx, info, infolen) \
             EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
-                              EVP_PKEY_CTRL_HKDF_INFO, infolen, (void *)(info))
+                              EVP_PKEY_CTRL_KDF_INFO, infolen, (void *)(info))
 
-# define EVP_PKEY_CTX_hkdf_mode(pctx, mode) \
+# define EVP_PKEY_CTX_set_pbe_mode(pctx, mode) \
             EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
-                              EVP_PKEY_CTRL_HKDF_MODE, mode, NULL)
+                              EVP_PKEY_CTRL_KDF_MODE, mode, NULL)
 
```

Comment:
I'm not sure about the renaming of some of these macros. As I said I consider "pbe" to be a subset of "kdf" so a macro that may be valid for several KDF algorithms (which may also include PBE) would have "kdf" in its name. Something specific to PBE would have pbe in it. That would change EVP_PKEY_CTX_set1_pbe_salt to EVP_PKEY_CTX_set1_kdf_salt for example (salts can be set in PBE and none-PBE KDFs), to me it just wouldn't look right using the macro EVP_PKEY_CTX_set1_pbe_salt to set the salt for a KDF which is not a PBE algorithm.

Reply:
So we are waiting for the submitter to update these names?
I will provisionally update labels accordingly.